### PR TITLE
[web] fixing text editing for autofill with semantics

### PR DESCRIPTION
**Pull request by nturgut**
## Description

Fixes the issues around autofill+semantics.

## Related Issues

https://github.com/flutter/flutter/issues/60402
https://github.com/flutter/flutter/issues/60404

## Tests

I added unit tests to text_editing_test.dart. I also unskip autofill related webkit tests.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [contributor guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [C++, Objective-C, Java style guides] for the engine.
- [x] I read the [tree hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation.
- [x] All existing and new tests are passing.
- [x] I am willing to follow-up on review comments in a timely manner.

## Reviewer Checklist

- [ ] I have submitted any presubmit flakes in this PR using the [engine presubmit flakes form] before re-triggering the failure.

## Breaking Change

Did any tests fail when you ran them? Please read [handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[contributor guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[tree hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[engine presubmit flakes form]: https://forms.gle/Wc1VyFRYJjQTH6w5A

### CODE REVIEW

File: lib/web_ui/lib/src/engine/semantics/semantics.dart [hunk 1]
```diff
@@ -927,6 +927,7 @@ class SemanticsObject {
           ..height = '${rect.height}px';
       }
     } else {
+      // This condition creates a issue in scrolling.
```
Comment (ferhatb) [inline lib/web_ui/lib/src/engine/semantics/semantics.dart:930]: Please add more detail, what issue?

Comment (nturgut) [reply lib/web_ui/lib/src/engine/semantics/semantics.dart:930]: ah, thanks for noticing. I was planning to do that. I even added this note to not forget.

File: lib/web_ui/test/text_editing_test.dart [hunk 2]
```diff
@@ -375,18 +382,77 @@ void testMain() {
...
+        // TODO(nurhan): https://github.com/flutter/flutter/issues/50769
+        skip: browserEngine == BrowserEngine.edge);
+
     test('Does not accept dom elements of a wrong type', () {
       // A regular <span> shouldn't be accepted.
       final HtmlElement span = SpanElement();
       expect(
-        () => SemanticsTextEditingStrategy(HybridTextEditing(), span),
+        () => SemanticsTextEditingStrategy(
+            new SemanticsObject(5, null), HybridTextEditing(), span),
```
Comment (mdebbar) [inline lib/web_ui/test/text_editing_test.dart:448]: nit: `new` isn't required.

**Review by mdebbar — APPROVED** [review]
LGTM

File: lib/web_ui/test/text_editing_test.dart (hunk 2 above)
Comment (nturgut) [reply lib/web_ui/test/text_editing_test.dart]: done

File: lib/web_ui/lib/src/engine/semantics/semantics.dart (hunk 1 above)
Comment (nturgut) [reply lib/web_ui/lib/src/engine/semantics/semantics.dart]: done.

**Pull request merged**